src/VisualStudio/Core/Def/Implementation/ProjectSystem/FileChangeTracker.cs

```diff
@@ -100,6 +108,27 @@ public void StartFileChangeListeningAsync()
             }
         }
 
+        private static bool ShouldTrapException(Exception e)
+        {
+            if (e is FileNotFoundException)
+            {
+                // The IVsFileChange implementation shouldn't ever be throwing exceptions like this, but it's a
+                // transient file system issue (perhaps the file being deleted while we're changing subscriptions)
+                // and so there's nothing better to do. We'll still non-fatal to track the rate this is happening
+                return FatalError.ReportWithoutCrash(e);
```

Comment:
ReportWithoutCrash is documented as "returning true". Figured if we make some central change that said "no, really don't catch exception type X" we get the benefit.